app/controllers/languages_controller.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# LanguagesController: for language functionality
+class LanguagesController < ApplicationController
+  def index; end
+
+  # Returns all languagaes as an array of arrays where each inner is in the format
+  # ['eng', 'English', supported: {sms: true, phone: true, email: true}]
+  # Example Return: [["eng", "English", {}]...,["zho","Chinese", {}],["zul","Zulu", {}]]
+  def language_data
+    render json: Languages.all_languages.map { |k, v| [k.to_sym, v[:display], v[:supported]] }
```

Comment:
Depend on how often this data is requested it would make sense to just create this data and serve the pre-built object.

Reply:
+1

Reply:
refactored this.